tools/parse/src/main/scala/org/allenai/nlpstack/parse/poly/fsm/FSMTrainingVectorSource.scala

```diff
@@ -34,7 +36,7 @@ abstract class FSMTrainingVectorSource(
     if (!vectorIter.hasNext) {
       resultSoFar
     } else {
-      taskHelper(resultSoFar + vectorIter.next.task, vectorIter)
+      taskHelper(resultSoFar + vectorIter.next().task, vectorIter)
     }
```

Comment:
Maybe this is relevant: The output of `Iterator.map` is another iterator. Calling `map` on an iterator does not pull in the entire contents of the iterator.